include/envoy/http/header_map.h

```diff
@@ -729,12 +743,55 @@ template <CustomInlineHeaderRegistry::Type type> class CustomInlineHeaderBase {
   }
 };
 
+#elif HEADER_MAP_TYPE == HEADER_MAP_TYPE_FLAT_HASH_MAP
+/**
+ * The following functions allow O(1) access for custom inline headers.
+ * Note: the simple-map implementation doesn't support this, and so an exception
+ * is thrown in this case (Not EnvoyException because using it requires changing
+ * the BUILD file and allowing code_format see the change).
+ */
+template <CustomInlineHeaderRegistry::Type type> class CustomInlineHeaderBase {
+public:
+  virtual ~CustomInlineHeaderBase() = default;
+
+  static constexpr CustomInlineHeaderRegistry::Type header_map_type = type;
+  using Handle = CustomInlineHeaderRegistry::Handle<header_map_type>;
+
+  virtual const HeaderEntry* getInline(Handle) const {
+    throw std::runtime_error("CustomInlineHeader isn't supported as part of simple-map");
```

Comment:
note: if you want to get the entire test suite passing I think you can very easily support this by just pulling the header string from the handle and looking it up, etc.

Reply:
I agree, it can probably be done the same way you implemented `TypedHeaderMapImpl`.
IMHO, if there will be no inline headers, then a simpler API should probably be introduced.

Reply:
Yes I agree that if we get rid of inline headers we can simplify the API, however a) it's not clear to me that we don't end up with different impls with different tradeoffs and b) in the interim it would be useful to actually be able to test something out with the existing code.